clis/teliod/src/daemon.rs

```diff
@@ -47,32 +65,112 @@ fn telio_task(
         None,
     )?;
 
+    // Keep track of current meshnet states
+    let mut state = TelioTaskStates::default();
+    // Mock the interface configuration for tests
+    let sys_config: Box<dyn SystemConfigurator> = if auth_token.as_str().eq(EMPTY_TOKEN) {
+        // And some fake IP address
+        state.interface_ip_address = Some(IpAddr::V4(Ipv4Addr::new(100, 100, 100, 100)));
+        Box::new(MockSystemConfigurator::new(false))
+    } else {
+        Box::new(IfconfigSystemConfigurator)
+    };
```

Comment:
auth token is defining if we're running tests?

wouldn't conditional compilation fit?

Reply:
or as said above, we can use generics instead

Reply:
The tests are not the main topic of this PR, however I don't see why we define a different behaviour for integration tests. Production code should not "be aware" if we are doing any integration tests. Therefore, I think this should be removed and if there's any mocking to be done should be to help unit testing.

Reply:
Moved these out of this PR